vault/readSecret_test.go

```diff
@@ -24,3 +24,55 @@ func TestReadSecretWrongPath(t *testing.T) {
 	// assert
 	assert.Equal(t, fmt.Sprintf(secretNotFound, path, nil), err.Error())
 }
+
+func testReadSecretKey(path string, key string, expectedValue interface{}, t *testing.T) {
+	// mock the ReadSecret function
+	vaultClient := &API{
+		Client: testVault.Client,
+	}
+
+	// act
+	value, err := vaultClient.ReadSecretKey(path, key)
+
+	// assert
+	assert.NilError(t, err)
+	assert.Equal(t, expectedValue, value)
+}
+
+// TestReadSecretKeyWithNumberAsValue tests that the function returns the correct secret key
+func TestReadSecretKeyWithNumberAsValue(t *testing.T) {
+	// arrange
+	path := "secret/data/secret"
+	key := "key4"
+	expectedValue := float64(123)
+
+	testReadSecretKey(path, key, expectedValue, t)
+}
+
+// TestReadSecretKeyWithBooleanAsValue tests that the function returns the correct secret key
```

Comment:
These comments might as well not be there 😄

Reply:
Yeah forgot to update them :D
Should be fixed now